### PR TITLE
queue restart ice offer while publisher pc gathering ice

### PR DESCRIPTION
**Pull request by cnderrauber**
If client send a offer to restart publisher pc ice while it is gathering ice candidate, queue the offer before gather complete.

### CODE REVIEW

**Review by boks1971 — APPROVED** [review]
Nice!

File: pkg/rtc/transport.go [hunk 1]
```diff
@@ -337,12 +352,37 @@ func (t *PCTransport) Close() {
...
+
+	var (
+		iceCredential   string
+		offerRestartICE bool
+	)
+	if sd.Type == webrtc.SDPTypeOffer {
+		var err error
+		iceCredential, offerRestartICE, err = t.isRemoteOfferRestartICE(sd)
+		if err != nil {
+			t.Logger().Errorw("check remote offer restart ice failed", err)
```
Comment (davidzhao) [inline pkg/rtc/transport.go:364]: should we still set the remote description anyways? thinking if there's a scenario where parsing failed, but the offer is actually valid.

File: pkg/rtc/transport.go [hunk 2]
```diff
@@ -337,12 +352,37 @@ func (t *PCTransport) Close() {
...
+		t.lock.Unlock()
+		return nil
+	}
 
 	if err := t.pc.SetRemoteDescription(sd); err != nil {
+		t.lock.Unlock()
 		return err
 	}
 
+	if t.currentOfferIceCredential == "" || offerRestartICE {
```
Comment (davidzhao) [inline pkg/rtc/transport.go:382]: should these checks all move into the `sd.Type == webrtc.SDPTypeOffer` block?

File: pkg/rtc/transport.go (hunk 1 above)
Comment (cnderrauber) [reply pkg/rtc/transport.go:364]: pion do the parse in `SetRemoteDescription` too, it will fail if parse failed.

File: pkg/rtc/transport.go (hunk 2 above)
Comment (cnderrauber) [reply pkg/rtc/transport.go:382]: we only save `iceCredential` if SetRemoteDescription success.

**Pull request merged**